src/validators/common.ts

```diff
@@ -0,0 +1,241 @@
+export const validateIdentifiers: ValidateIdentifiersFunc = _.partial(
+	validateMultiple, _.partial.placeholder,
+	validateIdentifier, _.partial.placeholder
+);
+
+export function validateNameSectionName(value: any): void {
+	validateRequiredString(value, 'nameSection.name');
+}
+
+export function validateNameSectionSortName(value: any): void {
+	validateRequiredString(value, 'nameSection.sortName');
+}
+
+export function validateNameSectionLanguage(value: any): void {
+	validatePositiveInteger(value, 'nameSection.language', true);
+}
+
+export function validateNameSectionDisambiguation(value: any): void {
+	validateOptionalString(value, 'nameSection.disambiguation');
+}
+
+export function validateNameSection(
+	values: any
+): void {
+	validateNameSectionName(get(values, 'name', null));
+	validateNameSectionSortName(get(values, 'sortName', null));
+	validateNameSectionLanguage(get(values, 'language', null));
+	validateNameSectionDisambiguation(get(values, 'disambiguation', null));
+}
+
+export function validateSubmissionSectionNote(value: any): void {
+	validateOptionalString(value, 'submissionSection.note');
+}
+
+export function validateSubmissionSectionAnnotation(value: any): void {
+	validateOptionalString(value, 'submissionSection.annotation.content');
+}
+
+export function validateSubmissionSection(
+	data: any
+): void {
+	validateSubmissionSectionNote(get(data, 'note', null));
+	validateSubmissionSectionAnnotation(get(data, 'annotation.content', null));
+}
+
+export function validateAuthorCreditRow(row: any): void {
+	validateUUID(getIn(row, ['author', 'id'], null), 'authorCredit.author.id', true);
+	validateRequiredString(get(row, 'name', null), 'authorCredit.name');
+	validateOptionalString(get(row, 'joinPhrase', null), 'authorCredit.joinPhrase');
+}
+
+export const validateAuthorCreditSection = _.partialRight(
+	// Requires at least one Author Credit row or zero in case of optional
+	validateMultiple, _.partialRight.placeholder,
+	validateAuthorCreditRow, null, _.partialRight.placeholder
+);
+
+// In the merge editor we use the authorCredit directly instead of the authorCreditEditor state
+export function validateAuthorCreditSectionMerge(authorCredit: any): void {
+	validatePositiveInteger(get(authorCredit, 'id', null), 'authorCredit.id', true);
+}
+
+export type NameSection = {
+	name: string;
+	sortName: string;
+	language: number;
+	default?: boolean;
+	primary: boolean;
+	disambiguation?: string;
+};
+
+// TODO: Change website and validators to use `languageId` instead of `language` to make code cleaner?
+export type AliasSection = Record<string, AliasWithDefaultT & {
+	language: number;
+}>;
+
+// TODO: Change website and validators to use `typeId` instead of `type` to make code cleaner?
+export type IdentifierSection = Record<string, IdentifierT & {
+	type: number;
+}>;
+
+export type AuthorCreditRow = {
+	author: EntityStub;
+	joinPhrase?: string;
+	name: string;
```

Comment:
In bb-site the type also defines `position`, `authorBBID` and `authorCreditID` props. However, these are not present when creating/editing an entity, only when viewing it.
I suppose considering the use of these validators we can ignore it for now, although I wonder if that will affect you in the near future when you start parsing and consuming edition entities.
Let's revisit it then.
https://github.com/metabrainz/bookbrainz-site/blob/2315cb7707a31dc096758c03d7df128f15588d1e/src/client/entity-editor/author-credit-editor/actions.ts#L44-L51

Reply:
Good point, I was unsure about the exact need for this type definition as well.
Adding it was mostly an afterthought as all other sections had a type definition and I wanted the basic type to be there in the ORM without having to do another release in case I need this type soon.